tests/TestCase/I18n/TimeTest.php

```diff
@@ -433,55 +436,33 @@ public function testNice($class)
     public function testI18nFormat($class)
     {
         $time = new $class('Thu Jan 14 13:59:28 2010');
+
+        // Test the default format which shuld be SHORT for Time and FrozenTime
         $result = $time->i18nFormat();
         $expected = '1/14/10, 1:59 PM';
         $this->assertTimeFormat($expected, $result);
 
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'es-ES');
-        $expected = 'jueves, 14 de enero de 2010, 13:59:28 (GMT)';
+        // Test with a custom timezone
+        $result = $time->i18nFormat('HH:mm:ss', 'Australia/Sydney');
+        $expected = '00:59:28';
         $this->assertTimeFormat($expected, $result);
 
+        // Test using a time-specific format
         $format = [\IntlDateFormatter::NONE, \IntlDateFormatter::SHORT];
         $result = $time->i18nFormat($format);
         $expected = '1:59 PM';
         $this->assertTimeFormat($expected, $result);
 
-        $result = $time->i18nFormat('HH:mm:ss', 'Australia/Sydney');
-        $expected = '00:59:28';
+        // Test using a specific format, timezone and locale
+        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'es-ES');
+        $expected = 'jueves, 14 de enero de 2010, 13:59:28 (GMT)';
         $this->assertTimeFormat($expected, $result);
 
+        // Test with custom default locale set
         $class::setDefaultLocale('fr-FR');
         $result = $time->i18nFormat(\IntlDateFormatter::FULL);
         $expected = 'jeudi 14 janvier 2010 13:59:28 UTC';
         $this->assertTimeFormat($expected, $result);
-
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'es-ES');
-        $expected = 'jueves, 14 de enero de 2010, 13:59:28 (GMT)';
-        $this->assertTimeFormat($expected, $result, 'Default locale should not be used');
-
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'fa-SA');
-        $expected = 'پنجشنبه ۱۴ ژانویهٔ ۲۰۱۰، ساعت ۱۳:۵۹:۲۸ GMT';
-        $this->assertTimeFormat($expected, $result, 'fa-SA locale should be used');
-
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'en-IR@calendar=persian');
-        $expected = 'Thursday, Dey 24, 1388 at 1:59:28 PM GMT';
-        $this->assertTimeFormat($expected, $result);
-
-        $result = $time->i18nFormat([\IntlDateFormatter::SHORT, \IntlDateFormatter::FULL], null, 'fa-IR@calendar=persian');
-        $expected = '۱۳۸۸/۱۰/۲۴،‏ ۱۳:۵۹:۲۸ GMT';
-        $this->assertTimeFormat($expected, $result);
-
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, null, 'en-KW@calendar=islamic');
-        $expected = 'Thursday, Muharram 29, 1431 at 1:59:28 PM GMT';
-        $this->assertTimeFormat($expected, $result);
-
-        $result = $time->i18nFormat(\IntlDateFormatter::FULL, 'Asia/Tokyo', 'ja-JP@calendar=japanese');
-        $expected = '平成22年1月14日木曜日 22時59分28秒 日本標準時';
```

Comment:
Should we have some tests left behind for non-gregorian calendars?

Reply:
Yea, we should since we parse it to set the traditional calendar flag.